prompting/organic/organic_scoring_prompting.py

```diff
@@ -0,0 +1,249 @@
+import time
+from functools import partial
+from typing import Any, Literal, Sequence
+
+import bittensor as bt
+import torch
+from organic_scoring import OrganicScoringBase
+from organic_scoring.synth_dataset import SynthDatasetBase
+from prompting.agent import HumanAgent
+from prompting.base.neuron import BaseNeuron
+from prompting.cleaners.cleaner import CleanerPipeline
+from prompting.dendrite import DendriteResponseEvent, SynapseStreamResult
+from prompting.forward import handle_response
+from prompting.llms.vllm_llm import vLLM_LLM
+from prompting.organic.organic_task import OrganicTask
+from prompting.organic.synth_organic_task import SynthOrganicTask
+from prompting.protocol import StreamPromptingSynapse
+from prompting.rewards.pipeline import RewardPipeline
+from prompting.rewards.reward import RewardResult
+from prompting.tasks.task import make_system_prompt
+from prompting.utils.logging import log_event
+from prompting.utils.uids import get_random_uids, get_uids
+from starlette.types import Send
+from typing_extensions import override
+
+
+class OrganicScoringPrompting(OrganicScoringBase):
+    def __init__(
+        self,
+        axon: bt.axon,
+        synth_dataset: SynthDatasetBase | Sequence[SynthDatasetBase],
+        trigger_frequency: float | int,
+        trigger: Literal["seconds", "steps"],
+        validator: BaseNeuron,
+        trigger_frequency_min: float | int = 2,
+        trigger_scaling_factor: float | int = 50,
+    ):
+        super().__init__(
+            axon=axon,
+            synth_dataset=synth_dataset,
+            trigger_frequency=trigger_frequency,
+            trigger=trigger,
+            trigger_frequency_min=trigger_frequency_min,
+            trigger_scaling_factor=trigger_scaling_factor,
+        )
+        self._val = validator
+        # Organic scoring reward pipeline.
+        self._reward_pipeline = RewardPipeline(
+            selected_tasks=[OrganicTask.name, SynthOrganicTask.name],
+            device=self._val.device,
+            available_tasks={
+                OrganicTask.name: OrganicTask,
+                SynthOrganicTask.name: SynthOrganicTask,
+            }
+        )
+
+    async def _priority_fn(self, synapse: StreamPromptingSynapse) -> float:
+        """Priority function for the axon"""
+        return 1000000.0
+
+    async def _blacklist_fn(self, synapse: StreamPromptingSynapse) -> tuple[bool, str]:
+        """Blacklist function for the axon"""
+        return False, ""
```

Comment:
Does this mean that every request to the validator has the same priority, and no requests are ever blacklisted? If we assume that the machine running the api will have keys, we can get creative and limit blacklist to everything that's not from the same hotkey.

Reply:
Added OTF hotkey to the axon whitelist
I think priority should be equal for all the incoming organic queries